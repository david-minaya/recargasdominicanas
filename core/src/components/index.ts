export * from './icon/icon.component';
export * from './option-icon/option-icon.component';
export * from './text/text.component';
export * from './image/image.component';
export * from './title/title.component';
export * from './outline-card/outline-card.component';
export * from './field-error/field-error.component';
export * from './text-field/text-field.component';
export * from './text-area/text-area.component';
export * from './select-option/select-option.component';
export * from './select/select.component';
export * from './autocomplete/autocomplete.component';
export * from './button/button.component';
export * from './fill-button/fill-button.component';
export * from './outline-button/outline-button.component';
export * from './error/error.component';
export * from './outline-item/outline-item.component';
export * from './detail-item/detail-item.component';
export * from './setting-item/setting-item.component';
export * from './transaction-item/transaction-item.component';
export * from './product-item/product-item.component';
export * from './option/option.component';
export * from './sales-report/sales-report.component';
export * from './brand/brand.component';
export * from './toolbar-title/toolbar-title.component';
export * from './toolbar/toolbar.component';
export * from './header/header.component';
export * from './navbar/navbar.component';
export * from './navbar-option/navbar-option.component';
export * from './menu/menu.component';
export * from './tab/tab.component';
export * from './tab-bar/tab-bar.component';
export * from './modal/modal.component';
export * from './modal-toolbar/modal-toolbar.component';
export * from './modal-content/modal-content.component';
export * from './modal-options/modal-options.component';
export * from './modal-actions/modal-actions.component';
export * from './base-modal/base-modal.component';
export * from './loading-modal/loading-modal.component';
export * from './receipt/receipt.component';
export * from './receipt-modal/receipt-modal.component';
export * from './switch/switch.component';
export * from './switch-case/switch-case.component';
export * from './cancel-transaction-modal/cancel-transaction-modal.component';
export * from './base-cell/base-cell.component';
export * from './cell/cell.component';
export * from './cell-button/cell-button.component';
export * from './cell-icon/cell-icon.component';
export * from './cell-image/cell-image.component';
export * from './outline-table/outline-table.component';
export * from './page-container/page-container.component';
export * from './page-content/page-content.component';
export * from './page-toolbar/page-toolbar.component';
export * from './table/table.component';
export * from './table-header/table-header.component';
export * from './table-row/table-row.component';
export * from './transaction-modal/transaction-modal.component';
export * from './confirm-transaction-modal/confirm-transaction-modal.component';
export * from './error-transaction-modal/error-transaction-modal.component';
export * from './drawer/drawer.component';
export * from './drawer-header/drawer-header.component';
export * from './drawer-item/drawer-item.component';
export * from './protected-route/protected-route.component';
export * from './float-button/float-button.component';
export * from './datetime/datetime.component';
export * from './app-toolbar/app-toolbar.component';
export * from './sales-report-modal/sales-report-modal.component';
export * from './toast/toast.component';
export * from './error-page/error-page.component';
export * from './progress/progress.component';
export * from './indeterminate-progress-bar/indeterminate-progress-bar.component';
export * from './alert/alert.component';
export * from './pagination/pagination.component';
export * from './text-item/text-item.component';
