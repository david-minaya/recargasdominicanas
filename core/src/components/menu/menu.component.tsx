import React, { ReactNode, MouseEvent, RefObject, useEffect, useState } from 'react';
import { createPortal } from 'react-dom';
import { CSSTransition } from 'react-transition-group';
import { Style, mergeStyle } from './menu.module.css';

interface Props {
  open: boolean;
  anchor: RefObject<HTMLElement>;
  top?: number;
  right?: number;
  children: ReactNode;
  style?: Style;
  autoWith?: boolean;
  autofocus?: boolean;
  onClose?: () => void;
}

export function Menu(props: Props) {

  const {
    open,
    anchor,
    top = 0,
    right = 0,
    autoWith = false,
    autofocus = true,
    children,
    style: customStyle,
    onClose,
  } = props;

  const style = mergeStyle(customStyle);
  const [ref] = useState<any>({});

  useEffect(() => {

    let intervalId: NodeJS.Timer | undefined;

    if (open) {

      let lastPosition: number;

      intervalId = setInterval(() => {

        if (!anchor.current) return;

        const rect = anchor.current.getBoundingClientRect();
        const currentPosition = rect.y;

        if (currentPosition !== lastPosition) {
          updateMenuPosition();
        }

        lastPosition = currentPosition;

      }, 100);
    }

    return () => {
      if (intervalId) {
        clearInterval(intervalId);
      }
    };
  }, [open]);

  function handleMenuRef(menu: HTMLDivElement | null) {
    ref.current = menu;
    updateMenuPosition();
  }

  function handleClick(event: MouseEvent<HTMLDivElement>) {
    onClose?.();
    event.stopPropagation();
  }

  function updateMenuPosition() {

    const menu = ref.current;
    const element = anchor.current;

    if (menu && element) {

      const elementRect = element.getBoundingClientRect();
      const menuRect = menu.getBoundingClientRect();
      const bodyWidth = document.body.clientWidth;
      const bodyHeight = document.body.clientHeight;
      const menuBottom = elementRect.top + menuRect.height + 12;

      if (autoWith) {
        menu.style.width = `${elementRect.width}px`;
      }

      menu.style.right = `${bodyWidth - (elementRect.right - right)}px`;
      
      menu.style.top = menuBottom < bodyHeight 
        ? `${elementRect.bottom + top}px`
        : `${elementRect.top - (menuRect.height + top)}px`;

      menu.style.visibility = 'visible';
      menu.focus();
    }
  }

  return createPortal(
    <CSSTransition
      in={open}
      unmountOnExit={true}
      classNames={style.animation}
      timeout={160}
      nodeRef={ref}>
      <div 
        tabIndex={autofocus ? 0 : undefined}
        onBlur={onClose} 
        className={style.container}
        ref={handleMenuRef}
        onClick={handleClick}>
        {children}
      </div>
    </CSSTransition>, 
    document.body
  );
}
