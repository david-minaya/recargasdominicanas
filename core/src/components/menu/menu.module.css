.container {
  width: 240px;
  visibility: hidden;
  position: fixed;
  z-index: 3;
  background-color: white;
  border-radius: var(--small-border-radius);
  box-shadow: 0px 1px 4px lightgray;
  outline: none;
  padding: 6px 0px;
}

.animation_enter {
  opacity: 0;
  transform: translateY(-8px);
}

.animation_enterActive {
  transition: all 160ms;
  opacity: 1;
  transform: translateY(0px);
}

.animation_exit {
  opacity: 1;
  transform: translateY(0px);
}

.animation_exitActive {
  transition: all 160ms;
  opacity: 0;
  transform: translateY(-8px);
}
