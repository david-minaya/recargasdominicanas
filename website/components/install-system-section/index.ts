export * from './install-system-section';
