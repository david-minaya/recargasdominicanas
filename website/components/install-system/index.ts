export { InstallSystem } from './install-system';
