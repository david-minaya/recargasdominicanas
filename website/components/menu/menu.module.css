.menu {
  display: flex;
  flex-flow: row wrap;
  justify-content: space-around;
  align-items: center;
  padding: 8px 0px;
  border-bottom: 1px solid lightgray;
}

.animation_enter {
  height: 0px;
  opacity: 0;
}

.animation_enterActive {
  height: 160px;
  opacity: 1;
  transition: height 0.2s ease-out, opacity 0.6s ease-out 0.3s;
}

.animation_exit {
  height: 160px;
  opacity: 1;
}

.animation_exitActive {
  height: 0px;
  opacity: 0;
  transition: opacity 0.2s ease-in, height 0.2s ease-in 0.2s;
}

.option {
  font-weight: 500;
  text-decoration: none;
  color: var(--accent-color);
  font-size: 17px;
  margin: 28px 12px 16px;
  -webkit-tap-highlight-color: transparent;
}

.button_base {
  width: 200px;
  border: 2px solid var(--accent-color);
  margin: 28px;
}

.button_base:hover {
  color: white;
  background-color: var(--accent-color);
}

@media (min-width: 449px) {

  .animation_enter {
    height: 0px;
  }
  
  .animation_enterActive {
    height: 72px;
  }
  
  .animation_exit {
    height: 72px;
  }
  
  .animation_exitActive {
    height: 0px;
  }

  .option {
    margin: 18px;
  }
  
  .button_base {
    margin: 16px;
  } 
}

@media (min-width: 800px) {

  .menu {
    width: 400px;
    justify-self: end;
    justify-content: space-between;
    padding: 0px;
    border: none;
  }

  .option {
    margin: 0px;
  }
  
  .button_base {
    margin: 0px;
  }
}
