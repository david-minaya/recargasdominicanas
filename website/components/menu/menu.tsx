import React from 'react';
import Router from 'next/router';
import Link from 'next/link';
import { CSSTransition } from 'react-transition-group';
import { Button } from '../button';
import { style } from './menu.module.css';

interface props {
  show?: boolean;
}

export function Menu({ show = true }: props) {
  return (
    <CSSTransition 
      in={show} 
      timeout={{ enter: 900, exit: 400 }} 
      unmountOnExit    
      classNames={style.animation}>
      <div className={style.menu}>
        <Link href='/faq'>
          <a className={style.option}>
            Preguntas frecuentes
          </a>
        </Link>
        <Button 
          style={style.button} 
          text='Solicitar Sistema' 
          onClick={() => Router.push('/request-system')}/>
      </div>
    </CSSTransition>
  );
}
