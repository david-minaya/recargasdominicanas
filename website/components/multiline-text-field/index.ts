export * from './multiline-text-field';
