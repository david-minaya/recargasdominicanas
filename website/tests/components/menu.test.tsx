import React from 'react';
import { shallow, mount } from 'enzyme';
import { Menu } from '../../components/menu';

it('render menu component', () => {
  const menu = mount(<Menu/>);
  const option = menu.find('.option');
  const button = menu.find('.button');
  expect(option.text()).toBe('Preguntas frecuentes');
  expect(button.text()).toBe('Solicitar Sistema');
});

it('snapshot', () => {
  const menu = shallow(<Menu/>);
  expect(menu).toMatchSnapshot();
});
