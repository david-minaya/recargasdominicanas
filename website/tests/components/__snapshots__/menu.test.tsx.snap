// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="menu"
>
  <Link
    href="/"
  >
    <a
      className="option"
    >
      Preguntas frecuentes
    </a>
  </Link>
  <Button
    customStyle={
      Object {
        "default": "button",
      }
    }
    onClick={[Function]}
    text="Solicitar Sistema"
  />
</div>
`;
